### DISCUSSION TITLE
How to subscribe / unsubscribe to an external thing when a trigger is tracked / untracked?

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [leptos-rs/leptos#2365] [Q&A]
For example, I want to access a BLE characteristic with leptos. When the value of a characteristic is tracked, I want to start BLE notifications for that characteristics. If the value stops being tracked, I want to stop notifications on the BLE characteristic.

**Comment by gbj** [comment]
Hm... I don't think there's a way to track whether something is being tracked, no. I wonder whether something like `on_cleanup` would be helpful for you (which runs whenever the parent effect re-runs), or just tracking a trigger that you track and then notify at the appropriate time.

**Reply by ChocolateLoverRaj** [reply]
I'm not sure what you mean by using `on_cleanup` to know if something is tracked or not. But this would be a very useful feature. Similar to how a [Mobx Atom](https://mobx.js.org/custom-observables.html#creating-custom-observables-) works

**Reply by gbj** [reply]
We might be talking past one another. Could you provide an example of the code you'd like to be able to write, but can't?

**Reply by ChocolateLoverRaj** [reply]
```rs
pub struct BleAtom {
  // BLE variables
}

impl Atom for BleAtom {
  fn on_tracked(&self) {
    // Start BLE notifications
  }

  fn on_untracked(&self) {
    // Stop BLE notifications
  }

  fn get(&self) {
    // Returns value
  }
}

let ble_signal_like = create_signal_from_atom(BleAttom::new())

// Whenever the signal is tracked, notifications are started
```

**Reply by gbj** [reply]
Sorry, I was unclear. I understand the interface you're looking for. What I was wondering was in what situation you would want to use this -- an example of the application code you would write that uses `on_tracked` or `on_untracked`, for example.

This is not currently possible. Implementing it would require some additional tracking machinery, or it may be a mismatch on some level with the way our reactive system works: when an effect re-runs, it unsubscribes from all signals and then resubscribes to the ones that are used on the next call, which allows dynamic dependency tracking. I haven't used MobX so have not encountered this particular feature before, and nobody's raised the question before, so I can't say I've thought that deeply about what the implementation would look like here.
